File: languages/csharp/BitwardenClient.cs
namespace Bitwarden.Sdk;

public sealed class BitwardenClient : IDisposable
{
    private readonly ProjectsClient _projectsClient;
    private readonly SecretsClient _secretsClient;
    private readonly CommandRunner _commandRunner;
    private readonly BitwardenSafeHandle _handle;

    public BitwardenClient(BitwardenSettings? settings = null)
    {
        var clientSettings = new ClientSettings
        {
            ApiUrl = settings is { ApiUrl: not null } ? settings.ApiUrl : "https://api.bitwarden.com",
            IdentityUrl = settings is { IdentityUrl: not null } ? settings.IdentityUrl : "https://identity.bitwarden.com",
            DeviceType = DeviceType.Sdk,
            UserAgent = "Bitwarden DOTNET-SDK"
        };

        _handle = BitwardenLibrary.Init(clientSettings.ToJson());
        _commandRunner = new CommandRunner(_handle);
        _projectsClient = new ProjectsClient(_commandRunner);
        _secretsClient = new SecretsClient(_commandRunner);
    }

    public ResponseForApiKeyLoginResponse? AccessTokenLogin(string accessToken)
    {
        var command = new Command() { AccessTokenLogin = new AccessTokenLoginRequest { AccessToken = accessToken } };
        return _commandRunner.RunCommand<ResponseForApiKeyLoginResponse>(command);
    }

    public ProjectsClient Projects()
    {
        return _projectsClient;
    }

    public SecretsClient Secrets()
    {
        return _secretsClient;
    }

    public void Dispose()
    {
        BitwardenLibrary.FreeMemory(_handle);
        _handle.Dispose();
    }
}

File: languages/php/src/BitwardenClient.php
<?php

namespace Bitwarden\Sdk;

use Bitwarden\Sdk\Schemas\ClientSettings;
use Bitwarden\Sdk\Schemas\DeviceType;
use JsonException;

class BitwardenClient
{
    private BitwardenLib $bitwarden_lib;

    private ClientSettings $clientSettings;

    public ProjectsClient $projects;

    public SecretsClient $secrets;

    public AuthClient $auth;

    private CommandRunner $commandRunner;

    /**
     * @throws JsonException
     */
    public function __construct(BitwardenSettings $bitwardenSettings)
    {
        $this->clientSettings = new ClientSettings(apiUrl: $bitwardenSettings->get_api_url(),
            deviceType: DeviceType::$SDK, identityUrl: $bitwardenSettings->get_identity_url(),
            userAgent: "Bitwarden PHP-SDK");

        $this->bitwarden_lib = new BitwardenLib();
        $this->bitwarden_lib->init($this->clientSettings);

        $this->commandRunner = new CommandRunner($this->bitwarden_lib);
        $this->projects = new ProjectsClient($this->commandRunner);
        $this->secrets = new SecretsClient($this->commandRunner);
        $this->auth = new AuthClient($this->commandRunner);
    }

    public function __destruct()
    {
        $this->bitwarden_lib->free_mem();
    }
}
